core/src/main/java/hudson/init/TaskMethodFinder.java

```diff
@@ -107,7 +108,19 @@ protected void invoke(Method e) {
         } catch (IllegalAccessException x) {
             throw (Error)new IllegalAccessError().initCause(x);
         } catch (InvocationTargetException x) {
-            throw new Error(x);
+            rethrow(x);
+        }
+    }
+
+    // Unwrap the useless InvocationTargetException and rethrow the cause as-is unless it is a checked exception
+    private void rethrow(InvocationTargetException x) {
+        Throwable cause = x.getCause();
+        if (cause instanceof RuntimeException) {
+            throw (RuntimeException) cause;
```

Comment:
is the casting necessary? it's a throwable?

Reply:
This is to rethrow an unchecked exception.

Reply:
I see 👍

Reply:
Java cannot prove the `Throwable` instance in not a checked exception - albeit we check it is not at runtime - so it insist it to be declared in the signature forcing the clients to handle `Throwable`.